crowbar_framework/app/models/node_object.rb

```diff
@@ -1451,6 +1451,26 @@ def disk_release!(device, owner)
     disk_release(device, owner) and save
   end
 
+  def check_unclaimed_disks
+    unclaimed_disks = Array.new
+    crowbar_wall[:claimed_disks].each do |disk, _claim|
+      path, device_name = disk.split("/")[-2..-1]
+
+      unless disk =~ /^\/dev\/disk\//
+        unclaimed_disks.push(device_name) unless node[:block_device][path].nil?
```

Comment:
Why not pushing the full disk name (```disk```) into the result array? That makes it easier to lookup the problematic claims for the calling code.

It would just need look into node[:claimed_disk][element_from_the_result] to figure out what claim is problematic.